hashedixsearch/search.py

```diff
@@ -126,7 +144,7 @@ def execute_query_exact(index, term):
 
 
 def ngram_to_term(ngram, stemmer, analyzer):
-    text = " ".join(ngram)
+    text = " ".join((token for token in ngram if token.strip()))
```

Comment:
This text filtering operation only seems necessary due to a mismatch between the whitespace tokenization of the `terms` provided to the `highlight` function and the tokenization of the `text` argument here.

Perhaps by enforcing / encouraging consistency between these, we can remove the need for this operation.